File: packages/rs-dpp/src/errors/dpp_error.rs
use thiserror::Error;

use crate::identity::errors::{AssetLockOutputNotFoundError, AssetLockTransactionIsNotFoundError};

#[derive(Error, Debug)]
#[error("{0}")]
pub enum DPPError {
    AssetLockOutputNotFoundError(AssetLockOutputNotFoundError),
    AssetLockTransactionIsNotFoundError(AssetLockTransactionIsNotFoundError),
    #[error("expected public key hash to be 20 bytes")]
    WrongPublicKeyHashSize,
    #[error("expected output type to be OP_RETURN output")]
    WrongBurnOutputType,
    #[error("invalid transaction")]
    InvalidAssetLockTransaction,
    #[error("core message corruption {0}")]
    CoreMessageCorruption(String),
}

File: packages/wasm-dpp/lib/errors/DPPError.ts
export class DPPError extends Error {
    name: string;
    message: string;

    constructor(message: string) {
        super();

        this.name = this.constructor.name;
        this.message = message;

        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, this.constructor);
        }
    }
}
